src/Controller/ClassController.php

```diff
@@ -142,7 +148,47 @@ private function considerClassificationStoreForColumnConfig(Request $request, ?D
             }
         }
 
-        if ($displayMode == ColumnConfigDisplayMode::ALL || ($displayMode == ColumnConfigDisplayMode::RELEVANT && !$enrichment)) {
+        if ($displayMode == ColumnConfigDisplayMode::ALL && $grouped === true) {
+            $class->setFieldDefinitions($fieldDefinitions);
+            $classString = 'Pimcore\\Model\\DataObject\\' . $class->getName();
```

Comment:
Using hardcoded namespaces could probably lead to problems, since they can be overwritten using `class_overrides`. 

We could inject the `FactoryInterface` here and build the object with the factory. Similar to what is done here: 

https://github.com/pimcore/pimcore/blob/3ffaeac8d9f07f877ea4f469015be1774de447b9/models/DataObject/AbstractObject.php#L244